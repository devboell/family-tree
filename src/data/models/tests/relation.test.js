import knex from '../../connectors'
import RelationModel from '../relation'

describe('RelationModel', () => {
  beforeEach(async () => {
    await knex.migrate.rollback()
    await knex.migrate.latest()
    await knex.seed.run()
  })

  afterEach(async () => {
    await knex.migrate.rollback()
  })

  afterAll(() => {
    knex.destroy()
  })

  it('findAll', async () => {
    const relations = await RelationModel.findAll()
    expect(relations).toMatchSnapshot()
  })

  it('findByPersonId', async () => {
    const relations = await RelationModel.findByPersonId(5)
    expect(relations).toMatchSnapshot()
  })

  it('create', async () => {
    const personId = 1
    const partnerids = [2, 3, 4]
    const relationIds = await RelationModel.create(personId, partnerids)
    expect(relationIds).toMatchSnapshot()
  })

  it('remove', async () => {
    await RelationModel.remove([1, 2, 3])
    const relations = await RelationModel.findAll()
    expect(relations).toMatchSnapshot()
  })
})
