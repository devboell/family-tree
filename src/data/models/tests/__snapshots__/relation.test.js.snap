// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RelationModel create 1`] = `
Array [
  9,
]
`;

exports[`RelationModel findAll 1`] = `
Array [
  Object {
    "id": 1,
    "person1Id": 1,
    "person2Id": 2,
  },
  Object {
    "id": 2,
    "person1Id": 4,
    "person2Id": 5,
  },
  Object {
    "id": 3,
    "person1Id": 7,
    "person2Id": 8,
  },
  Object {
    "id": 4,
    "person1Id": 9,
    "person2Id": 10,
  },
  Object {
    "id": 5,
    "person1Id": 4,
    "person2Id": 14,
  },
  Object {
    "id": 6,
    "person1Id": 15,
    "person2Id": 16,
  },
]
`;

exports[`RelationModel findByPersonId 1`] = `
Array [
  Object {
    "id": 2,
  },
]
`;

exports[`RelationModel remove 1`] = `
Array [
  Object {
    "id": 4,
    "person1Id": 9,
    "person2Id": 10,
  },
  Object {
    "id": 5,
    "person1Id": 4,
    "person2Id": 14,
  },
  Object {
    "id": 6,
    "person1Id": 15,
    "person2Id": 16,
  },
]
`;
